libraries/chain/controller.cpp

```diff
@@ -4840,14 +4840,20 @@ struct controller_impl {
       return conf.block_validation_mode == validation_mode::LIGHT || conf.trusted_producers.count(producer);
    }
 
+   int32_t max_reversible_blocks_allowed() const {
+      if (conf.max_reversible_blocks == 0)
+         return std::numeric_limits<int32_t>::max();
+      return conf.max_reversible_blocks - fork_db_savanna_size();
```

Comment:
shouldn't it also return std::numeric_limits<int32_t>::max() when we have a legacy fork_db instead of max_reversible_blocks?

Otherwise in net_plugin you could have `(reversible_remaining < sync_fetch_span)`. I guess maybe it is not a big deal,  it just may limit `sync_fetch_span` which could seem strange.

Reply:
Maybe. As is, if `max-reversible-blocks` is default 3600 it limits `sync-fetch-span` to 3600. I kind of thought that a good thing. But would be ok with changing it to return `MAX_INT32` if legacy fork db. @arhag what do you think?

Reply:
Personally, I think `std::numeric_limits<int32_t>::max()` if more bulletproof, as you don't have to check how the return value is used in net_plugin to know it won't add a limitation.

Reply:
I do think it should return the max value from this function if only dealing with the legacy fork db. It is also consistent with your existing comment describing `max_reversible_blocks_allowed` in `controller.hpp`.